File: LeetCode/Top Interview Questions/Easy Collection/Math/Count Primes.py
class Solution:
    def countPrimes(self, n: int) -> int:
        if n <= 2:
            return 0
        isPrime = [True] * n
        isPrime[0] = False
        isPrime[1] = False
        for i in range(2, n):
            if (isPrime[i] == True) and (i*i <= n):
                for j in range(i*i, n, i):
                    isPrime[j] = False
        answer = 0
        for element in isPrime:
            if element == True:
                answer += 1
        return answer

File: LeetCode/Top Interview Questions/Easy Collection/Math/Count Primes.cpp
class Solution {
public:
    int countPrimes(int n) {
        //Resolvendo para os casos base.
        if(n == 0 || n == 1){
            return 0;
        }
        //Resolvendo para o caso geral: n >= 2.
        vector<bool> isPrime(n, true); // Inicialmente, assumo que todos os numeros sao primos.
        int primeCounter = 0;
        
        for(int i = 2; i * i < n; i++){
            if(isPrime[i] == true){
                for(int j = 2; j * i < n; j++){
                    isPrime[i * j] = false;
                }
            }
        }
        
        for(int i = 2; i < n; i++){
            if(isPrime[i] == true) primeCounter++;
        }
        
        return primeCounter;
    }
};
